src/bun.js/test/expect.zig

```diff
@@ -57,8 +69,30 @@ pub const Expect = struct {
             none,
         } = .none,
         not: bool = false,
+
+        const BitsetType = i32;
+        const FLAG_NOT: BitsetType = (1 << 0);
+        const FLAG_RESOLVES: BitsetType = (1 << 1);
+        const FLAG_REJECTS: BitsetType = (1 << 2);
+
+        /// Used to pass the flags to C++ asymmetric matchers execution
+        /// Note: keep in sync with the corresponding C++ bitset flags
+        pub fn toBitset(this: Flags) BitsetType {
+            return (if (this.not) FLAG_NOT else 0) | (if (this.promise == .resolves) FLAG_RESOLVES else 0) | (if (this.promise == .rejects) FLAG_REJECTS else 0);
+        }
+
+        pub fn fromBitset(bitset: BitsetType) Flags {
+            return Flags{
+                .not = (bitset & FLAG_NOT) != 0,
+                .promise = if ((bitset & FLAG_RESOLVES) != 0) .resolves else if ((bitset & FLAG_REJECTS) != 0) .rejects else .none,
+            };
+        }
```

Comment:
Same comment, good candidate for packed structs.

Reply:
Indeed, this makes more sense here
Thanks